extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -229,4 +234,44 @@ private function isEuropaTheme() {
 
 		return !empty( $wgEnablePortableInfoboxEuropaTheme );
 	}
+
+	private function createSmartGroups( $groupData, $rowCapacity ) {
+		$result = [ ];
+		$capacity = 0;
+		$smartGroup = [ ];
+
+		foreach ( $groupData as $item ) {
+			$data = $item['data'];
+
+			if ( $item['type'] === 'data' && ( !isset( $data['layout'] ) || $data['layout'] !== 'default' ) ) {
+
+				if ( $capacity + $data['span'] > $rowCapacity ) {
+					$result = array_merge( $result, $this->calculateRowItemsSize( $smartGroup, $capacity ) );
+					$capacity = 0;
```

Comment:
I think 'currentSize' or 'slotsTaken' better fits to what this variable contains than 'capacity'